pkg/controller/logstash/logstash_controller_test.go

```diff
@@ -0,0 +1,224 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package logstash
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	logstashv1alpha1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/logstash/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/comparison"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/hash"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/watches"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
+)
+
+func newReconcileLogstash(objs ...runtime.Object) *ReconcileLogstash {
+	r := &ReconcileLogstash{
+		Client:         k8s.NewFakeClient(objs...),
+		recorder:       record.NewFakeRecorder(100),
+		dynamicWatches: watches.NewDynamicWatches(),
+	}
+	return r
+}
+
+func TestReconcileLogstash_Reconcile(t *testing.T) {
+	defaultLabels := (&logstashv1alpha1.Logstash{ObjectMeta: metav1.ObjectMeta{Name: "testLogstash"}}).GetIdentityLabels()
+	tests := []struct {
+		name     string
+		objs     []runtime.Object
+		request  reconcile.Request
+		want     reconcile.Result
+		expected logstashv1alpha1.Logstash
```

Comment:
Would it make sense to also check that all the other objects have been reconciled? I think only validating that they were created would be fine as their "contents" are verified in other tests.

For example:

```golang
type expectedObject struct {
	t    client.Object
	name types.NamespacedName
}

type expectedObjects []expectedObject

func (e expectedObjects) assertExist(t *testing.T, k8s client.Client) {
	t.Helper()
	for _, o := range e {
		obj := o.t.DeepCopyObject().(client.Object)
		assert.NoError(t, k8s.Get(context.Background(), o.name, obj), "Expected object not found: %s", o.name)
	}
}
```

And then in a test:

```golang
		{
			name: "Logstash with ready StatefulSet and Pod updates status properly",
			[...]
			expectedObjects: []expectedObject{
				{
					t:    &corev1.Service{},
					name: types.NamespacedName{Namespace: "test", Name: "testLogstash-ls-api"},
				},
				{
					t:    &corev1.Secret{},
					name: types.NamespacedName{Namespace: "test", Name: "testLogstash-ls-config"},
				},
			},
			wantErr: false,
		},
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
[...]
			tt.expectedObjects.assertExist(t, r.Client)
[...]
}
```

Reply:
Great idea! Done